build.js

```diff
@@ -466,131 +699,173 @@ function executeOnCreate(api, app, config, opts) {
+          "-keystore", keyPath, "-storepass", "android",
+          "-signedjar", "app-debug-aligned.apk",
+          "app-signed.apk", "androiddebugkey"
+        ];
+
+        alignArgsDebug = [
+          "-f", "-v", "4", "app-debug.apk", "app-debug-aligned.apk"
+        ];
+
+        var apkDirDebug = path.join(outputPath, "../../" + shortName + "/app/build/outputs/apk/debug/");
+        return spawnWithLogger(api, process.env.ANDROID_HOME + '/build-tools/27.0.3/zipalign', alignArgsDebug, {cwd: apkDirDebug})
+          .then(function () {
+            spawnWithLogger(api, process.env.ANDROID_HOME + '/build-tools/27.0.3/apksigner', signArgsDebug, {cwd: apkDirDebug})
+          })
+      }
+
+    } else {
+      var keystore = process.env['DEVKIT_ANDROID_KEYSTORE'];
+      if (!keystore) { throw new BuildError('missing environment variable DEVKIT_ANDROID_KEYSTORE'); }
 
-    var storepass = process.env['DEVKIT_ANDROID_STOREPASS'];
-    if (!storepass) { throw new BuildError('missing environment variable DEVKIT_ANDROID_STOREPASS'); }
+      var storepass = process.env['DEVKIT_ANDROID_STOREPASS'];
+      if (!storepass) { throw new BuildError('missing environment variable DEVKIT_ANDROID_STOREPASS'); }
 
-    var keypass = process.env['DEVKIT_ANDROID_KEYPASS'];
-    if (!keypass) { throw new BuildError('missing environment variable DEVKIT_ANDROID_KEYPASS'); }
+      var keypass = process.env['DEVKIT_ANDROID_KEYPASS'];
+      if (!keypass) { throw new BuildError('missing environment variable DEVKIT_ANDROID_KEYPASS'); }
 
-    var key = process.env['DEVKIT_ANDROID_KEY'];
-    if (!key) { throw new BuildError('missing environment variable DEVKIT_ANDROID_KEY'); }
+      var key = process.env['DEVKIT_ANDROID_KEY'];
+      if (!key) { throw new BuildError('missing environment variable DEVKIT_ANDROID_KEY'); }
 
-    signArgs = [
-        "-sigalg", "MD5withRSA", "-digestalg", "SHA1",
-        "-keystore", keystore, "-storepass", storepass, "-keypass", keypass,
-        "-signedjar", shortName + "-unaligned.apk",
-        shortName + "-release-unsigned.apk", key
+      signArgs = [
+        "sign", "--ks", keystore, "--ks-pass", "pass:"+storepass, "--key-pass", "pass:"+keypass,
+        "--ks-key-alias", key, "--v1-signing-enabled", "true", "--v2-signing-enabled", "false", "--verbose",
+        "app-release-aligned.apk" //shortName + "app-release-unsigned.apk", key
       ];
 
-    alignArgs = [
-        "-f", "-v", "4", shortName + "-unaligned.apk", shortName + "-aligned.apk"
+      alignArgs = [
+        "-f", "-v", "4", "app-release-unsigned.apk", "app-release-aligned.apk" //shortName + "-unaligned.apk", shortName + "-aligned.apk"
       ];
+
+      var scheme = (config.debug ? "debug" : "release");
+      var apkDir = path.join(outputPath, "../../"+shortName+"/app/build/outputs/apk/"+scheme+"/");
+      return spawnWithLogger(api, process.env.ANDROID_HOME +'/build-tools/27.0.3/zipalign', alignArgs , {cwd: apkDir})
```

Comment:
why hardcoded build-tools version? cant it be system specific version ?
build tool path should be added to PATH variable

Reply:
It should be same as tools used in gradle. This can be used in config and used in build.js and gradle synchronized.

Reply:
But when we are updating some plugins, it may required some particular build-tools version.

So hard coding here will cause issue in feature.

Reply:
This is being tracked here -> https://github.com/CodeCombiner/native-android/issues/64